### ISSUE TITLE
Wrapped line unwrapped and "line exceeded maximum width"

### ISSUE THREAD

**Issue by mverleg** [rust-lang-nursery/rustfmt#2610]
I have an issue with a wrapped line getting unwrapped and then being too long:

    lazy_static! {
        static ref INTERNER: Mutex<StringInterner<usize, RandomState>> =
            Mutex::new(StringInterner::new());
    }

This `static ref` thing gets reformatted to one line, which is then too long.

I did `rustup component add rustfmt-preview --toolchain nightly` just today.

Maybe related to #2067 ?

**Comment by topecongiro** [comment]
rustfmt does not respect the width of indent inside `lazy_static!`:

https://github.com/rust-lang-nursery/rustfmt/blob/518ece227223cd6fc6d808c893b4404f450261f2/src/macros.rs#L1297

We should replace `block_indent` with `block_left`, and add a test in lazy_static.rs:

```diff
- let nested_shape = shape.block_indent(context.config.tab_spaces()); 
+ let nested_shape = shape.block_left(context.config.tab_spaces()); 
```

**Comment by SvenKnabe** [comment]
I would like to work at this one as my first issue.

**Comment by nrc** [comment]
@SvenKnabe great! https://github.com/rust-lang-nursery/rustfmt/issues/2610#issuecomment-379967321 gives instructions for how to fix this, let us know if you have any questions once you get stuck in.

**Comment by dsrees** [comment]
@nrc the value of `block_left` will return an `Option<Shape>` where `block_indent` returns only a `Shape`. What would be the best approach to handle the `Option` inside of `format_lazy_static`?

**Comment by topecongiro** [comment]
@dsrees We often use `?` to handle this kind of stuff in rustfmt. When we want to continue formatting even when there is no width left, we use `.unwrap_or` and retry in different way (e.g. use newline).

Anyhow, this issue is fixed via #2681, so I am going to close this. Thanks for your interests in contributing to rustfmt!
